src/main/java/controllers/MemberController.java

```diff
@@ -174,4 +172,108 @@ public void updateProfile(UserData data){
             throw new RuntimeException(e);
         }
     }
-}
+
+    /**
+     * Updates the fitness metric for either goal or current stats
+     * Inserts if no record exists yet.
+     * @param stats list of stats
+     * @return true if successfully added, false otherwise
+     */
+    public boolean updateFitnessMetrics(List<Object> stats) {
```

Comment:
if you change the parameter type you can remove all the instanceof Integer checks in the method.
```suggestion
    public boolean updateFitnessMetrics(List<Integer> stats) {
```